docs/source/testing/testing.md

```diff
@@ -3,26 +3,55 @@ title: Integration testing
+Integration testing a GraphQL server means testing the combination of many interlocking parts at once. One way to test your server setup is to use `ApolloServer`'s `executeOperation` method to directly execute a GraphQL operation without going through a full HTTP operation.
+
+## Testing using `executeOperation`
+
+The `apollo-server` library has a request pipeline that can support many plugins which can each affect the way an operation is executed. The `executeOperation` method is available to run operations through the request pipeline, enabling the most thorough tests possible without starting up an HTTP server. 
+
+The [`executeOperation` method](../api/apollo-server/#executeoperation) can accept the following arguments: 
+* An object containing configuration options which must include a `query` key specifying the GraphQL operation to be run. 
+  * You can use `executeOperation` to execute both queries and mutations, but both use the `query` key. 
+* An argument that is passed in to the `ApolloServer` instance's [`context` function](../data/resolvers/#the-context-argument).
+
+Below is a simplified example of setting up a test using the JavaScript testing library [Jest](https://jestjs.io/):
+```js:title=index.test.js
+// For clarity in this example we included our typeDefs and resolvers above our test,
+// but in a real world situation you'd be importing these in
+const typeDefs = gql`
+  type Query {
+    hello(name: String): String!
+  }
+`;
+
+const resolvers = {
+  Query: {
+    hello: (_, { name }) => `Hello ${name}!`,
+  },
+};
+
+it('returns hello with the provided name', async () => {
+  const mockServer = new ApolloServer({
+    typeDefs,
+    resolvers
+  });
 
-```javascript
-const server = new ApolloServer(config);
+  const result = await mockServer.executeOperation({
+    query: 'query SayHelloWorld($name: String) { hello(name: $name) }',
+    variables: { name: 'world' },
+  });
 
-const result = await server.executeOperation({
-  query: GET_USER,
-  variables: { id: 1 }
+  expect(result.errors).toBeUndefined();
+  expect(result.data?.hello).toBe('Hello world!');
 });
-expect(result.errors).toBeUndefined();
-expect(result.data?.user.name).toBe('Ida');
 ```
+Note that when testing, any errors in parsing, validating, and executing your GraphQL operation are returned in the `errors` field of the result rather than thrown (just like a normal GraphQL response).
 
-For example, you can set up a full server with your schema and resolvers and run an operation against it.
+Unlike with a normal instance of `ApolloServer`,  you don't need to call [`start`](../../api/apollo-server/#start) before calling `executeOperation`. The server instance will call `start` automatically for you if it has not been called, and any startup errors will be thrown.
```

Comment:
"will call" -> "calls"

"if it has not been called" -> "if it hasn't been called already"

"will be thrown" -> "are thrown"